### DISCUSSION TITLE
Preact compatibility issue

### DISCUSSION THREAD

**Discussion by LME10** [vercel/swr#1645] [Q&A]
I tried using useSWR in preact 10.5 version but ended up in below error.

`Uncaught TypeError: Cannot read properties of undefined (reading 'context')
    at F (index.js?a178:238)
    at Z (index.esm.js?4b4e:1)`
    
    Please could you suggest work around for this issue

**Comment by huozhi** [comment]
Hi, can you provide a reproduction that how you encounter with this issue?

**Reply by huozhi** [reply]
@LME10  I can run preact with latest swr well in [this codesandbox](https://codesandbox.io/s/preact-swr-upbeat-frost-jv6n8-jv6n8) without errors. Could you provide the full trace or a reproduction for it?

**Reply by laymonage** [reply]
Hey @huozhi, someone made a reproduction here:

https://github.com/preactjs/next-plugin-preact/issues/53#issuecomment-1075552659

I also forked it to use Preact without next-plugin-preact, same issue: https://codesandbox.io/s/nextjs-swr-preact-forked-8cl57b

**Comment by pc-erin** [comment]
With 2.0 it seems like SWR has an explicit dependency on React.

Is there a way to make it depend on react or preact/hooks depending on what is available?

**Comment by bklynate** [comment]
I ran into a similar issue. +1
